### DISCUSSION TITLE
openvpn bei 7362SL Fehler

### DISCUSSION THREAD

**Discussion by hurzl** [Freetz-NG/freetz-ng#823] [Allgemein] [answered]
```
XXX/freetz-ng/toolchain/build/mips_gcc-5.5.0_uClibc-1.0.14-nptl_kernel-3.10/mips-linux-uclibc/bin/mips-linux-uclibc-gcc -W -Wall -Wshadow -Wundef -D_GNU_SOURCE -march=34kc -mtune=34kc -msoft-float -Os -pipe -Wa,--trap -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o .libs/pscap pscap.o  -L../src XXX/freetz-ng/source/target-mips_gcc-5.5.0_uClibc-1.0.14-nptl_kernel-3.10/libcap-ng-0.8.3/src/.libs/libcap-ng.so
XXX/freetz-ng/source/target-mips_gcc-5.5.0_uClibc-1.0.14-nptl_kernel-3.10/libcap-ng-0.8.3/src/.libs/libcap-ng.so: undefined reference to `pthread_atfork'
collect2: error: ld returned 1 exit status
make[4]: *** [Makefile:480: captest] Fehler 1
```

Linux bharch 6.4.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 27 Jul 2023 22:02:18 +0000 x86_64 GNU/Linux

Einzige geänderte Option.

**Comment by kollb69** [comment]
Hallo, ich habe die gleiche Box und unter Ubuntu 22.04.3 tritt ebenfalls der libcap-ng Fehler auf. Folgender Workaround aus diesem Ticket [https://github.com/stevegrubb/libcap-ng/issues/9](url) funktioniert bei mir. Erstmal mit `make libcap-ng-dirclean` das Verzeichnis bereinigen, dann mit dem zusätzlichen Befehl  `make LDFLAGS="-lpthread" libcap-ng-precompiled` durchlaufen lassen. Bei Erfolg kann dann einfach mit `make` weitergemacht werden. Dem Compiler scheint hier den zusätzlichen Befehl zu benötigen. In der freetz Oberfläche wird dann im erstellten Image auch die neue OpenVPN-Version angezeigt. Müsste dann jemand mal testen ob das dann damit auch funktioniert.

**Comment by feedzapper** [comment]
Ihr müsst vor dem compilieren unter den TOOLCHAIN OPTIONS -> Target binutils (binutils-2.41) wählen
Weiterhin am Besten :
Patchelf version (old)  ---> 
[ ] Use precompiled x86_64 host-tools

**Reply by fda77** [reply]
- old patchelf gibt es nur für leute die noch keinen system-gcc7 haben, das steht aber auch in der hilfe
 - mit target-gcc5 gibt es diese binutils version nicht

**Comment by fda77** [comment, marked as answer]
@kollb69 ich hab  pthread einfach ausgemacht, compiliert auch 24ea4f7f1
